### ISSUE TITLE
training loss becomes NaN

### ISSUE THREAD

**Issue by HustLion** [Thund3rPat/kohya_ss-linux#11]
Hello, thank you for your excellent port to linux.

I got the training working following your instructions. But after about 100 steps of training, the loss becomes NaN and the final result is not usable (won't be loaded by webui when generating images).

the loss display:

![image](https://user-images.githubusercontent.com/12825646/220552120-10793ad4-beac-4031-9552-d711783b5822.png)

I found out that the newest code might be the cause that breaks the training. The issue is here: https://github.com/bmaltais/kohya_ss/issues/215

I checked the issue and `git checkout 72c0cb7f632ab56a786b4c83e01de012ceaff96c` to the corresponding commit in this repo but I still get NaN. 
![image](https://user-images.githubusercontent.com/12825646/220552938-3cae800a-820d-4763-a03d-498cba54240a.png)

Any ideas on how to deal with this?

**Comment by Thund3rPat** [comment]
Hello, I have also encountered this problem. However, it is difficult for me to determine what may be the cause.
I can provide you with my config that I have had success with. 
Please check if you have success with these parameters.

```json
{
  "pretrained_model_name_or_path": "",
  "v2": false,
  "v_parameterization": false,
  "logging_dir": "",
  "train_data_dir": "",
  "reg_data_dir": "",
  "output_dir": "",
  "max_resolution": "512,512",
  "learning_rate": "0.0001",
  "lr_scheduler": "constant",
  "lr_warmup": "0",
  "train_batch_size": 1,
  "epoch": "1",
  "save_every_n_epochs": "1",
  "mixed_precision": "fp16",
  "save_precision": "fp16",
  "seed": "1234",
  "num_cpu_threads_per_process": 2,
  "cache_latents": false,
  "caption_extension": ".txt",
  "enable_bucket": true,
  "gradient_checkpointing": false,
  "full_fp16": false,
  "no_token_padding": false,
  "stop_text_encoder_training": 0,
  "use_8bit_adam": true,
  "xformers": true,
  "save_model_as": "safetensors",
  "shuffle_caption": false,
  "save_state": false,
  "resume": "",
  "prior_loss_weight": 1.0,
  "text_encoder_lr": "5e-5",
  "unet_lr": "0.0001",
  "network_dim": 4,
  "lora_network_weights": "",
  "color_aug": false,
  "flip_aug": false,
  "clip_skip": 2,
  "gradient_accumulation_steps": 1.0,
  "mem_eff_attn": true,
  "output_name": "noshirov3",
  "model_list": "custom",
  "max_token_length": "75",
  "max_train_epochs": "",
  "max_data_loader_n_workers": "1",
  "network_alpha": 4,
  "training_comment": "",
  "keep_tokens": "0",
  "lr_scheduler_num_cycles": "",
  "lr_scheduler_power": "",
  "persistent_data_loader_workers": false,
  "bucket_no_upscale": true,
  "random_crop": false,
  "bucket_reso_steps": 64.0,
  "caption_dropout_every_n_epochs": 0.0,
  "caption_dropout_rate": 0,
  "optimizer": "AdamW"
}
```

**Comment by HustLion** [comment]
> Hello, I have also encountered this problem. However, it is difficult for me to determine what may be the cause. I can provide you with my config that I have had success with. Please check if you have success with these parameters.
> 
> ```json
> {
>   "pretrained_model_name_or_path": "",
>   "v2": false,
>   "v_parameterization": false,
>   "logging_dir": "",
>   "train_data_dir": "",
>   "reg_data_dir": "",
>   "output_dir": "",
>   "max_resolution": "512,512",
>   "learning_rate": "0.0001",
>   "lr_scheduler": "constant",
>   "lr_warmup": "0",
>   "train_batch_size": 1,
>   "epoch": "1",
>   "save_every_n_epochs": "1",
>   "mixed_precision": "fp16",
>   "save_precision": "fp16",
>   "seed": "1234",
>   "num_cpu_threads_per_process": 2,
>   "cache_latents": false,
>   "caption_extension": ".txt",
>   "enable_bucket": true,
>   "gradient_checkpointing": false,
>   "full_fp16": false,
>   "no_token_padding": false,
>   "stop_text_encoder_training": 0,
>   "use_8bit_adam": true,
>   "xformers": true,
>   "save_model_as": "safetensors",
>   "shuffle_caption": false,
>   "save_state": false,
>   "resume": "",
>   "prior_loss_weight": 1.0,
>   "text_encoder_lr": "5e-5",
>   "unet_lr": "0.0001",
>   "network_dim": 4,
>   "lora_network_weights": "",
>   "color_aug": false,
>   "flip_aug": false,
>   "clip_skip": 2,
>   "gradient_accumulation_steps": 1.0,
>   "mem_eff_attn": true,
>   "output_name": "noshirov3",
>   "model_list": "custom",
>   "max_token_length": "75",
>   "max_train_epochs": "",
>   "max_data_loader_n_workers": "1",
>   "network_alpha": 4,
>   "training_comment": "",
>   "keep_tokens": "0",
>   "lr_scheduler_num_cycles": "",
>   "lr_scheduler_power": "",
>   "persistent_data_loader_workers": false,
>   "bucket_no_upscale": true,
>   "random_crop": false,
>   "bucket_reso_steps": 64.0,
>   "caption_dropout_every_n_epochs": 0.0,
>   "caption_dropout_rate": 0,
>   "optimizer": "AdamW"
> }
> ```

Thank you! I will try this config. Currently I moved on to windows platform with A100 card. But there's no luck either. The training on A100 just hangs on caching latents or the first step of training. I'm inspecting kohya source code to determine if A100 card is not supported by kohya yet.

**Comment by HustLion** [comment]
I'm testing the master version of this repo now. Before this test, I tried another version with your config.

My process:

- `git clone https://github.com/Thund3rPat/kohya_ss-linux` install per your intructions
- `cd kohya_ss-linux`
- `pip install --use-pep517 --upgrade -r requirements.txt`
- `git checkout a008c62893720825a1de134eb615a30d7e7a512c` this is the commit where is said to have no NaN issue
- manually modify lora_gui.py `subprocess.run(run_cmd)` to `subprocess.run(run_cmd, shell=True)`
- run `python ./kohya_gui.py`

And the set the training according to your config. My final config is:

```json
{
  "pretrained_model_name_or_path": "/root/autodl-tmp/sd-model/v1-5-pruned-emaonly.ckpt",
  "v2": false,
  "v_parameterization": false,
  "logging_dir": "/root/autodl-tmp/datasets/ynxy/log",
  "train_data_dir": "/root/autodl-tmp/datasets/ynxy/img4",
  "reg_data_dir": "",
  "output_dir": "/root/autodl-tmp/datasets/ynxy/model",
  "max_resolution": "512,512",
  "learning_rate": "0.0001",
  "lr_scheduler": "constant",
  "lr_warmup": "0",
  "train_batch_size": 1,
  "epoch": "1",
  "save_every_n_epochs": "1",
  "mixed_precision": "fp16",
  "save_precision": "fp16",
  "seed": "1234",
  "num_cpu_threads_per_process": 2,
  "cache_latents": false,
  "caption_extension": ".txt",
  "enable_bucket": true,
  "gradient_checkpointing": false,
  "full_fp16": false,
  "no_token_padding": false,
  "stop_text_encoder_training": 0,
  "use_8bit_adam": true,
  "xformers": true,
  "save_model_as": "safetensors",
  "shuffle_caption": false,
  "save_state": false,
  "resume": "",
  "prior_loss_weight": 1.0,
  "text_encoder_lr": "5e-5",
  "unet_lr": "0.0001",
  "network_dim": 4,
  "lora_network_weights": "",
  "color_aug": false,
  "flip_aug": false,
  "clip_skip": 2,
  "gradient_accumulation_steps": 1.0,
  "mem_eff_attn": true,
  "output_name": "ynxy_style_v1.6",
  "model_list": "custom",
  "max_token_length": "75",
  "max_train_epochs": "",
  "max_data_loader_n_workers": "1",
  "network_alpha": 4,
  "training_comment": "",
  "keep_tokens": "0",
  "lr_scheduler_num_cycles": "",
  "lr_scheduler_power": "",
  "persistent_data_loader_workers": false,
  "bucket_no_upscale": true,
  "random_crop": false,
  "bucket_reso_steps": 64.0,
  "caption_dropout_every_n_epochs": 0.0,
  "caption_dropout_rate": 0
}
```

This is apparently an old version of config since there's no `"optimizer": "AdamW"`.

With this, I got 1600 steps of no NaN, but after that, NaN apprears again. But this is by far the best result I got with linux version. Before this, I can only train to 100 steps.

Here's the loss image (best result is 20230223110129/network_train):

![image](https://user-images.githubusercontent.com/12825646/220815110-bfbcbadc-aea7-4cd3-a71f-d712ef21d3f6.png)

**Comment by HustLion** [comment]
Using the HEAD version also stops at around 1300 steps and turns NaN.
